rust/arrow/src/compute/kernels/boolean.rs

```diff
@@ -123,9 +125,78 @@ pub fn not(left: &BooleanArray) -> Result<BooleanArray> {
     Ok(BooleanArray::from(Arc::new(data)))
 }
 
+pub fn is_null(input: &ArrayRef) -> Result<BooleanArray> {
+    if input.offset() % 8 != 0 {
```

Comment:
Is this limitation fine, or is there a potential way around it? Does it mean that `array[len = 30].slice(5, 20).is_null()` would fail?

Reply:
This is currently a limitation of several kernels that operate on boolean arrays or null bitmaps because we try to operate on whole bytes (or larger) instead of individual bits for performance reasons. We would need a nicer abstraction for those bit packed buffers to iterate over chunks, regardless of offsets. I think it makes sense to look into such an implementation together with @jorgecarleitao proposal for an iterator interface.

Reply:
Oh yes, I remember that now. I tried looking at bit manipulation that would address this, but I can't remember where; maybe as part of the Parquet writer 🤔

It's def something that we need to do, if I have more time in the coming months I'll look into it. We could also take inspiration from c++ as I think they have this.